lib/wallet.js

```diff
@@ -315,6 +315,144 @@ class WalletClient extends Client {
     return this.get(`/wallet/${id}/coin/${hash}/${index}`);
   }
 
+  /**
+   * Get name state for the given name.
+   * {@see hsd.NameState}
+   * @param {String} id
+   * @param {String} name
+   * @returns {Promise}
+   */
+
+  getName(id, name) {
+    return this.get(`/wallet/${id}/name/${name}`);
+  }
+
+  /**
+   * Get name state for all names
+   * that the wallet is managing.
+   * {@see hsd.NameState}
+   * @param {String} id
+   * @param {Object} options
+   * @returns {Promise}
+   */
+
+  getNames(id, options) {
+    return this.get(`/wallet/${id}/name`, options);
+  }
+
+  /**
+   * Get bids, reveals and name state
+   * for the given name.
+   * {@see hsd.NameState}
+   * {@see hsd.BlindBid}
+   * {@see hsd.BidReveal}
+   * @param {String} id
+   * @param {String} name
+   * @returns {Promise}
+   */
+
+  getAuctionByName(id, name) {
+    return this.get(`/wallet/${id}/auction/${name}`);
+  }
+
+  /**
+   * Get bids, reveals and name state
+   * for all names the wallet manages.
+   * {@see hsd.NameState}
+   * {@see hsd.BlindBid}
+   * {@see hsd.BidReveal}
+   * @param {String} id
+   * @param {Object} options
+   * @returns {Promise}
+   */
+
+  getAuctions(id, options) {
+    return this.get(`/wallet/${id}/auction`, options);
+  }
+
+  /**
+   * Get wallet bids for a given name.
```

Comment:
nit: I do not think we need to qualify the bids as "wallet bids." What do you think?

Reply:
I see what you mean, I was following this prior art - https://github.com/handshake-org/hsd/blob/master/lib/wallet/http.js#L667 

Do you think that people will understand that the wallet does not index all bids for all names? That is why I qualified the bids as "wallet bids", to make that point clear. Maybe its not needed or there is a better way to word it

Reply:
Pushed up a fix to remove the word wallet